joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForStatementsCreator.scala

```diff
@@ -258,6 +259,11 @@ trait AstForStatementsCreator(implicit withSchemaValidation: ValidationMode) { t
         astsForStatement(node).toList
   }
 
+  private def returnAstForRubyCall[C <: RubyCall](node: RubyNode with RubyCallWithBlock[C]): Seq[Ast] = {
+    val Seq(methodDecl, typeDecl, methodRef, callAst) = astsForCallWithBlock(node)
+    methodDecl :: typeDecl :: methodRef :: returnAst(returnNode(node, code(node)), List(callAst)) :: Nil
```

Comment:
The `methodDecl` and `typeDecl` will automatically be linked to the parent method via `AstLinkerPass`, and the `callAst` should have the `methodRef` as an argument. 
```suggestion
    returnAst(returnNode(node, code(node)), List(callAst)) :: Nil
```